2019-04-29  Senyang Jiang  <classsen@lnxsrv06.seas.ucla.edu>

        * timespec.h (timespec_cmp): changed implementation of the function to prevent integer overflow


--- timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec_corrected.h	2019-04-29 14:37:30.582660000 -0700
@@ -45,8 +45,15 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if(a.tv_sec < b.tv_sec){
+    return -1;
+  }
+  else if(a.tv_sec > b.tv_sec){
+    return 1;
+  }
+  else{
+    return a.tv_nsec > b.tv_nsec ? 1 : (a.tv_nsec < b.tv_nsec ? -1 : 0);
+  }
 }
 
 # if ! HAVE_DECL_NANOSLEEP
